src/SprykerEco/Zed/AvalaraTax/Business/Checker/AvalaraTaxQuoteChecker.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\AvalaraTax\Business\Checker;
+
+use Generated\Shared\Transfer\CheckoutErrorTransfer;
+use Generated\Shared\Transfer\CheckoutResponseTransfer;
+use Generated\Shared\Transfer\MessageTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use SprykerEco\Zed\AvalaraTax\Dependency\Facade\AvalaraTaxToCalculationFacadeInterface;
+
+class AvalaraTaxQuoteChecker implements AvalaraTaxQuoteCheckerInterface
+{
+    /**
+     * @var \SprykerEco\Zed\AvalaraTax\Dependency\Facade\AvalaraTaxToCalculationFacadeInterface
+     */
+    protected $calculationFacade;
+
+    /**
+     * @param \SprykerEco\Zed\AvalaraTax\Dependency\Facade\AvalaraTaxToCalculationFacadeInterface $calculationFacade
+     */
+    public function __construct(AvalaraTaxToCalculationFacadeInterface $calculationFacade)
+    {
+        $this->calculationFacade = $calculationFacade;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\CheckoutResponseTransfer $checkoutResponseTransfer
+     *
+     * @return bool
+     */
+    public function isQuoteTaxCalculationValid(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponseTransfer): bool
+    {
+        $quoteTransfer = $this->calculationFacade->recalculateQuote($quoteTransfer);
```

Comment:
* I think we don't need to recalculate a quote here.
* After adding an item to the cart, recalculation is always called.

Reply:
Resolved.